raiden/network/proxies/token_network.py

```diff
@@ -251,19 +245,35 @@ def new_netting_channel(
             )
             raise InvalidSettleTimeout(msg)
 
-        if not self.client.can_query_state_for_block(given_block_identifier):
-            raise NoStateForBlockIdentifier(
-                "Tried to open a channel with a block identifier older than "
-                "the pruning limit. This should not happen."
+        # check preconditions
+        try:
+            existing_channel_identifier = self.get_channel_identifier_or_none(
+                participant1=self.node_address,
+                participant2=partner,
+                block_identifier=given_block_identifier,
             )
-
-        channel_exists = self._channel_exists_and_not_settled(
-            participant1=self.node_address,
-            participant2=partner,
-            block_identifier=given_block_identifier,
-        )
-        if channel_exists:
-            raise DuplicatedChannelError("Channel with given partner address already exists")
+            balance = self.token.balance_of(
+                address=to_checksum_address(self.address), block_identifier=given_block_identifier
+            )
+            limit = self.token_network_deposit_limit(block_identifier=given_block_identifier)
+            safety_deprecation_switch = self.safety_deprecation_switch(given_block_identifier)
+        except ValueError:
+            # If `given_block_identifier` has been pruned the checks cannot be
+            # performed.
+            pass
+        except BadFunctionCallOutput:
+            raise_on_call_returned_empty(given_block_identifier)
+        else:
+            if existing_channel_identifier is not None:
+                raise BrokenPreconditionError(
+                    "A channel with the given partner address already exists."
+                )
+            if balance >= limit:
+                raise BrokenPreconditionError(
+                    "Cannot open another channel, token network deposit limit has been reached."
+                )
+            if safety_deprecation_switch:
+                raise BrokenPreconditionError("This token network is deprecated.")
 
         # Prevent concurrent attempts to open a channel with the same token and
         # partner address.
```

Comment:
Humm, could you also remove replace `open_channel_transactions` with `self.channel_operations_lock[partner]`?

The pattern we are using make the thing with an `AsyncResult` unnecessary (But it may change behavior from returning the channel id to raising a `Recoverable` exception)